payment-request/PaymentRequestUpdateEvent/updateWith-call-immediate-manual.https.html

```diff
@@ -0,0 +1,171 @@
+  requestShipping: true,
+});
+
+function testImmediateUpdate({ textContent: testName }) {
+  promise_test(async t => {
+    const request = new PaymentRequest(
+      validMethods,
+      validDetails,
+      validOptions
+    );
+    const shippingOptionEventPromise = new Promise((resolve, reject) => {
+      request.addEventListener(
+        "shippingaddresschange",
+        async ev => {
+          // spin the event loop, sets [[waitForUpdate]] to true.
+          await Promise.resolve();
+          try {
+            ev.updateWith(validDetails);
+            resolve(); // This is bad.
+          } catch (err) {
+            reject(err); // this is good.
+          }
+        },
+        { once: true }
+      );
+    });
+    const response = await request.show();
+    await promise_rejects(
+      t,
+      "InvalidStateError",
+      shippingOptionEventPromise,
+      "The event loop already spun, so [[waitForUpdate]] is now true"
+    );
+    await response.complete();
+  }, testName.trim());
+}
+
+function testSubsequentUpdateWithCalls({ textContent: testName }) {
+  promise_test(async t => {
+    const request = new PaymentRequest(
+      validMethods,
+      validDetails,
+      validOptions
+    );
+    const eventPromise = new Promise((resolve, reject) => {
+      request.addEventListener("shippingaddresschange", async ev => {
+        const p = Promise.resolve(validDetails);
+        ev.updateWith(p);
+        await p;
+        try {
+          ev.updateWith(validDetails);
+          resolve(); // this is bad, we should never get to here.
+        } catch (err) {
+          reject(err); // this is good!
+        }
+      });
+    });
+    const responsePromise = request.show();
+    await promise_rejects(t, "InvalidStateError", eventPromise);
+    const response = await responsePromise;
+    await response.complete();
+  }, testName.trim());
+}
+
+function testRecycleEvents({ textContent: testName }) {
+  promise_test(async t => {
+    const request = new PaymentRequest(
+      validMethods,
+      validDetails,
+      validOptions
+    );
+    const addressChangedPromise = new Promise(resolve => {
+      request.addEventListener("shippingaddresschange", resolve, {
+        once: true,
+      });
+    });
+    const optionChangedPromise = new Promise(resolve => {
+      request.addEventListener("shippingoptionchange", resolve, { once: true });
+    });
+
+    const responsePromise = request.show();
+
+    // Let's wait for the address to change.
+    const addressChangeEvent = await addressChangedPromise;
+    addressChangeEvent.updateWith(validDetails);
+
+    // Let's wait for the shippingOption
+    const optionChangeEvent = await optionChangedPromise;
+
+    // The addressChangeEvent was already consumed and expired.
+    assert_throws(
+      new TypeError(),
+      () => {
+        addressChangeEvent.updateWith(validDetails);
+      },
+      "addressChangeEvent [[waitingForUpdate]] is true, so it must throw"
```

Comment:
I think addressChangeEvent.[[WaitForUpdating]] is false after optionChangeEvent.updateWith(), because the updating of the first addChangeEvent.updateWith() must be processed before selecting an option and emitting the optionChangeEvent. Maybe we need to update the error message to the correct situation.
I am not sure what kind of error it should be, TypeError, InvalidStateError, or others.

Reply:
> I think addressChangeEvent.[[WaitForUpdating]] is false, 

We changed the spec so `[[waitingForUpdate]]`  can't ever go back to false. Once .updateWith() is called, it's always true.

Reply:
> I am not sure what kind of error it should be, TypeError, InvalidStateError, or others.

The error should have been `InvalidStateError` - so good catch!